configure.ac

```diff
@@ -889,6 +889,44 @@ AC_SUBST(OPENTHREAD_NCP_VENDOR_HOOK_SOURCE)
 AC_DEFINE_UNQUOTED([OPENTHREAD_ENABLE_NCP_VENDOR_HOOK],[${OPENTHREAD_ENABLE_NCP_VENDOR_HOOK}],[Define to 1 if using NCP vendor hook])
 AM_CONDITIONAL([OPENTHREAD_ENABLE_NCP_VENDOR_HOOK], [test "${OPENTHREAD_ENABLE_NCP_VENDOR_HOOK}" = "1"])
 
+#
+# NCP Spinel Transformer
+#
+
+AC_ARG_WITH(
+    [ncp-spinel-transformer-libs],
+    [AS_HELP_STRING([--with-ncp-spinel-transformer-libs=<LIBSPINEL_TRANSFORMER.A>],[Specify library files (absolute paths) implementing the NCP Spinel Transformer. @<:@default=none@:>@.])],
+    [
+        if test "${withval}" = "no"
+        then OPENTHREAD_ENABLE_NCP_SPINEL_TRANSFORMER=0
+        else
+            OPENTHREAD_ENABLE_NCP_SPINEL_TRANSFORMER=1
+            OPENTHREAD_NCP_SPINEL_TRANSFORMER_LIBS=${withval}
+        fi
+    ],
+    [
+        OPENTHREAD_ENABLE_NCP_SPINEL_TRANSFORMER=0
+    ])
+
+AC_MSG_CHECKING([for NCP Spinel Transformer])
+AC_MSG_RESULT(${OPENTHREAD_NCP_SPINEL_TRANSFORMER_LIBS-none})
+AC_SUBST(OPENTHREAD_NCP_SPINEL_TRANSFORMER_LIBS)
+AC_DEFINE_UNQUOTED([OPENTHREAD_ENABLE_NCP_SPINEL_TRANSFORMER],[${OPENTHREAD_ENABLE_NCP_SPINEL_TRANSFORMER}],[Define to 1 if using NCP Spinel Transformer])
+AM_CONDITIONAL([OPENTHREAD_ENABLE_NCP_SPINEL_TRANSFORMER], [test "${OPENTHREAD_ENABLE_NCP_SPINEL_TRANSFORMER}" = "1"])
+
+AC_ARG_WITH(
+    [ncp-spinel-transformer-outbound-buffer-size],
+    [AS_HELP_STRING([--with-ncp-spinel-transformer-outbound-buffer-size=<SIZE IN BYTES>],[Specify a NCP Spinel Transformer's outbound buffer size. @<:@default=none@:>@.])],
```

Comment:
I suggest for this we instead use a define in `openthread-core-default-config.h` file  like `OPENTHREAD_CONFIG_NCP_SPINEL_TRANSFORMER_BUFFER_SIZE`. 

The overall idea is to use `configure.ac` to enable/disable (bigger) features, and keep all parameters which impact behavior in the above config header file.

Reply:
Thanks for the review. We added option "--with-ncp-spinel-transformer-outbound-buffer-size" because it's strictly related to specific library used to transform spinel frames (added with option "--with-ncp-spinel-transformer-libs"). IMO, it's nice to have the ability to control this whole feature in the same way (and in one place), i.e. using aforementioned options. What do you think?

Reply:
The reason we adopted this model is mainly to ensure `configrue.ac` does not become too big with too many options. We have adopted/used this approach for many existing features already (for example, `--enable-mac-filter` in `configure.ac` but `OPENTHREAD_CONFIG_MAC_FILTER_SIZE` defined in header file, same for jam-detection, child-supervision, ncp configs, etc.).  

Personally I prefer header file `#define`s since we can keep/see the documentation/description of the parameter in comments next to definition.